src/MonthSelect.php

```diff
@@ -4,24 +4,84 @@
 
 namespace Laminas\Filter;
 
+use function filter_var;
+use function is_array;
+use function is_numeric;
+use function sprintf;
+
+use const FILTER_VALIDATE_INT;
+
 /**
  * @psalm-type Options = array{
  *     null_on_empty?: bool,
  *     null_on_all_empty?: bool,
- *     ...
  * }
  * @psalm-type InputArray = array{
- *       year: numeric,
- *       month: numeric,
+ *       year: int,
+ *       month: int,
  *  }
- * @template TOptions of Options
- * @template-extends AbstractDateDropdown<TOptions, InputArray>
+ * @template TInput of array<array-key, numeric>
+ * @implements FilterInterface<string>
  */
-final class MonthSelect extends AbstractDateDropdown
+final class MonthSelect implements FilterInterface
 {
-    /**
-     * Year-Month
-     */
-    protected string $format      = '%2$s-%1$s';
-    protected int $expectedInputs = 2;
+    private readonly bool $returnNullIfAnyFieldEmpty;
+    private readonly bool $returnNullIfAllFieldsEmpty;
+
+    /** @param Options $options */
+    public function __construct(array $options = [])
+    {
+        $this->returnNullIfAnyFieldEmpty  = $options['null_on_empty'] ?? false;
+        $this->returnNullIfAllFieldsEmpty = $options['null_on_all_empty'] ?? false;
+    }
+
+    public function __invoke(mixed $value): mixed
+    {
+        return $this->filter($value);
+    }
+
+    /** @psalm-suppress InvalidReturnType */
+    public function filter(mixed $value): mixed
+    {
+        if (! is_array($value)) {
+            return $value;
+        }
+
+        /** @var mixed $month */
+        $month = $value['month'] ?? null;
+        /** @var mixed $year */
+        $year = $value['year'] ?? null;
+
+        if ($this->returnNullIfAnyFieldEmpty && ($month === null || $year === null)) {
+            return null;
+        }
+
+        if ($this->returnNullIfAllFieldsEmpty && $month === null && $year === null) {
+            return null;
+        }
+
+        if (! $this->isParsableAsDateValue($month, 1, 12) || ! $this->isParsableAsDateValue($year, 0, 9999)) {
+            /** @psalm-suppress InvalidReturnStatement */
+            return $value;
+        }
+
+        /** @psalm-suppress MixedArgument */
+        return sprintf('%s-%s', $year, $month);
+    }
+
+    private function isParsableAsDateValue(mixed $value, int $lowestValue, int $highestValue): bool
```

Comment:
I was hoping we could annotate this with `@psalm-assert-if-true int<T1, T2> $value` but it's not possible: https://psalm.dev/r/4a9988cce3

Alternatively, you can annotate as `@psalm-assert-if-true int $value`, that will fix the MixedArgument issue in `filter` I'd expect.